[bdAd] Exclude Pages

maitandat1507 opened the pull request:
# Cho phép chọn các page không hiển thị quảng cáo

maitandat1507 committed:
[bdAd] Exclude Pages
# Cho phép chọn các page không hiển thị quảng cáo

daohoangson commented on library/bdAd/Listener.php:
```diff
@@ -77,4 +79,27 @@ public static function helperEngine()
+        $routeMatch->getMajorSection();
+        $majorSection = $routeMatch->getMajorSection();
+
+        $listValuesOption = XenForo_Application::getOptions('bdAd_customizeDisplayAd')->getOptions('bdAd_customizeDisplayAd');
```
Xài `bdAd_Option::get()` nha.

daohoangson commented on library/bdAd/Listener.php:
```diff
@@ -77,4 +79,27 @@ public static function helperEngine()
+        foreach ($checkedValues as $key => $value) {
+            if ($majorSection == $key) {
+                self::$_noAd = false;
+                self::bdAd_customizeDisplayAdCheck();
```
Gọi hàm này ở đây làm gì vậy?

maitandat1507 replied:
À do lúc đầu em làm nhầm **$_noAd == false**, thấy không chạy nên thử bỏ thêm vô thôi, để em xóa ạ.

daohoangson commented on library/bdAd/addon-bdAd.xml:
```diff
@@ -577,6 +578,14 @@
   <email_template_modifications/>
   <optiongroups>
     <group group_id="bdAd" display_order="9999" debug_only="0"/>
+    <option option_id="bdAd_customizeDisplayAd" edit_format="checkbox" data_type="array" can_backup="1">
```
Bỏ lựa chọn `search` đi nếu chưa hỗ trợ.

maitandat1507 committed:
[bdAd] Exclude Pages
# Edit some code in bdAd_Listener.php
# Delete "search" in bdAd_customizeDisplayAd option

daohoangson commented on library/bdAd/Listener.php:
```diff
@@ -77,4 +79,25 @@ public static function helperEngine()
+
+    }
+
+    public static function bdAd_customizeDisplayAdCheck()
```
Tên hàm này có vấn đề:
1. Tại sao cần prefix "bdAd_"? Nó là hàm trong `bdAd_Listener` rồi mà?
2. Tên hàm nên giải thích xem nó làm gì, ví dụ tên hàm "customize xyz" thì sẽ hiểu là gọi hàm này để biến đổi hiển thị cái gì đó. Nhưng hóa ra gọi hàm này để kiểm tra xem có hiện ads không -> không liên quan tí nào.
3. Thường với hàm kiểm tra flag boolean thì nên để prefix là "is", ví dụ hàm này có thể đặt là "isNoAd" hoặc "isAdFree". Tương tự, cái flag cũng sẽ đặt tên tương ứng với cái method "isNoAd" -> "$_noAd", "isAdFree" -> "$_adFree".

daohoangson commented on library/bdAd/addon-bdAd.xml:
```diff
@@ -577,6 +578,13 @@
   <email_template_modifications/>
   <optiongroups>
     <group group_id="bdAd" display_order="9999" debug_only="0"/>
+    <option option_id="bdAd_customizeDisplayAd" edit_format="checkbox" data_type="array" can_backup="1">
```
Tương tự comment về tên method ở trên, tên option này cũng không hiểu gì cả. Có thể đổi lại là "bdAd_noAdPages" hoặc "bdAd_adFreePages" etc.

daohoangson commented on library/bdAd/Listener.php:
```diff
@@ -40,7 +41,8 @@ public static function template_post_render_PAGE_CONTAINER(
         XenForo_Template_Abstract $template
-    ) {
+    )
+    {
```
Dòng 44 với 45 này commit thừa huh?

maitandat1507 committed:
[bdAd] Exclude Pages
# Rename method and variable
# Edit xml file, tạo mảng chứa

maitandat1507 commented:
l

maitandat1507 committed:
[bdAd] Exclude Pages
# Edit a few code
